nevergrad/optimization/optimizerlib.py

```diff
@@ -2411,3 +2411,56 @@ def recommend(self) -> p.Parameter:
 @registry.register
 class NGOpt(NGOpt10):
     pass
+
+
+class _MSR(CM):
+    """This code applies 16 copies of NGOpt10 with random weights for the different objective functions.
+
+    Variants dedicated to multiobjective optimization by multiple monoobjective optimization.
+    """
+
+    def __init__(
+        self,
+        parametrization: IntOrParameter,
+        budget: tp.Optional[int] = None,
+        num_workers: int = 1,
+        num_msr: int = 9,
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        assert budget is not None
+        self.num_optims = num_msr
+        self.optims = [
+            NGOpt(self.parametrization, budget=1 + (budget // self.num_optims), num_workers=num_workers)
+            for _ in range(self.num_optims)
+        ]
+        self.coeffs = [
+            self.parametrization.random_state.uniform(size=self.num_objectives)
+            for _ in range(self.num_optims)
+        ]
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        for i, opt in enumerate(self.optims):
+            this_loss = np.sum(loss * self.coeffs[i])
+            opt.tell(candidate, this_loss)
+
+
+class MultipleSingleRuns(base.ConfiguredOptimizer):
+    """CMA-ES optimizer,
+    Parameters
+    ----------
+    num_msr: int
+        number of single runs.
```

Comment:
please use a more explicit name.
Also, should it be possible to provide the name of the optimizer to use?

Reply:
Fixed + done.